src/software/new_geom/point.h

```diff
@@ -0,0 +1,451 @@
+#pragma once
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/polygon/point_concept.hpp>
+#include <cmath>
+#include <iostream>
+
+#include "software/new_geom/angle.h"
+#include "software/new_geom/vector.h"
+
+/**
+ * A point in 2D space.
+ */
+class Point final
+{
+   public:
+    /**
+     * Creates a Point at the origin (0, 0).
+     */
+    explicit constexpr Point();
+
+    /**
+     * Creates a unit-magnitude Point from an angle.
+     *
+     * @param angle the angle
+     *
+     * @return Point the Point
+     */
+    static Point createFromAngle(const Angle &angle);
+
+    /**
+     * Creates a Point at arbitrary coordinates.
+     *
+     * @param x the <var>x</var> value of the Point
+     * @param y the <var>y</var> value of the Point
+     */
+    constexpr Point(double x, double y);
+
+    /**
+     * Creates a new Point that is a copy of the given Point
+     *
+     * @param the Point to duplicate
+     */
+    constexpr Point(const Point &p);
+
+    /**
+     * Creates a new Point from a Vector
+     *
+     * @param the Vector to create a Point from
+     */
+    constexpr Point(const Vector &v);
+
+    /**
+     * Returns the x coordinate of this Point
+     *
+     * @return the x coordinate of this Point
+     */
+    constexpr double x() const;
+
+    /**
+     * Returns the y coordinate of this Point
+     *
+     * @return the y coordinate of this Point
+     */
+    constexpr double y() const;
+
+    /**
+     * Sets the coordinates of this point to the new coordinates
+     *
+     * @param x the new x coordinate
+     * @param y the new y coordinate
+     */
+    void set(double x, double y);
+
+    /**
+     * Sets the x coordinate of this point
+     *
+     * @param x the new x coordinate
+     */
+    void setX(double x);
+
+    /**
+     * Sets the y coordinate of this point
+     *
+     * @param y the new y coordinate
+     */
+    void setY(double y);
+
+    /**
+     * Returns the distance between this Point and origin (0,0)
+     *
+     * @return the distance
+     */
+    double distanceFromOrigin() const;
+
+    /**
+     * Returns the distance between this Point and the given point p
+     *
+     * @return the distance
+     */
+    double distanceFromPoint(const Point &p) const;
+
+    /**
+     * Returns a new Vector from this Point
+     *
+     * @return A new vector from this Point
+     */
+    Vector toVector() const;
+
+    /**
+     * Returns the unit vector in direction of this point
+     *
+     * @return A unit vector in the direction of this Point, or a
+     * zero-vector if this Point is zero
+     */
+    Vector norm() const;
```

Comment:
Does it make sense to normalize a Point? Are we intending to return a `Vector` here?